internal/apis/configuration/v1/types.go

```diff
@@ -111,6 +117,25 @@ type KongPlugin struct {
 	Protocols []string `json:"protocols,omitempty"`
 }
 
+// SecretValueFromSource represents the source of a secret value
+type SecretValueFromSource struct {
+	// the secret containing the key
+	Secret string `json:"name,omitempty"`
+	// the key containing the value
+	Key string `json:"key,omitempty"`
+}
```

Comment:
I might be missing something here but these Go types don't reflect what we have in the CRD.
I read this as `.configFrom.name`. The CRD has `.configFrom.secretKeyRef.name`.

Reply:
Annoying--thought I had this but apparently accidentally reverted it at some point. The tests don't catch this because we write the structs directly. Is there any way to mock up loading an actual YAML definition of the various resources?

Reply:
No. The way to solve this is to use kube-builder to automatically generate the CRDs rather than tweaking them by hand.